tests/app/main/test_suppliers.py

```diff
@@ -0,0 +1,216 @@
+import mock
+from mock import Mock
+from nose.tools import assert_equal, assert_true, assert_in, assert_false
+from tests.app.helpers import BaseApplicationTest
+
+
+def get_supplier(*args, **kwargs):
+    return {"suppliers": {
+        "id": 1234,
+        "description": "Supplier Description",
+        "clients": ["Client One", "Client Two"],
+        "contactInformation": [{
+            "id": 2,
+            "website": "supplier.dmdev",
+            "email": "supplier@user.dmdev",
+            "contactName": "Supplier Person",
+            "phoneNumber": "0800123123",
+            "address1": "1 Street",
+            "address2": "2 Building",
+            "city": "Supplierville",
+            "postcode": "11 AB",
+        }]
+    }}
+
+
+class TestSuppliersDashboard(BaseApplicationTest):
+    @mock.patch("app.main.views.suppliers.data_api_client")
+    def test_shows_supplier_info(self, data_api_client):
+        data_api_client.get_supplier.side_effect = get_supplier
+        with self.app.test_client():
+            self.login()
+
+            res = self.client.get("/suppliers")
+            assert_equal(res.status_code, 200)
+
+            data_api_client.get_supplier.assert_called_once_with(1234)
+
+            resp_data = res.get_data(as_text=True)
+
+            assert_in("Supplier Description", resp_data)
```

Comment:
Might it be worth testing that each field is in the right place on the page, rather than just somewhere on the page?  Could be a fiddly though.